source/v2/I2C.cpp

```diff
@@ -129,7 +129,14 @@ I2C::I2C(PinName sda, PinName scl) :
     // Select the appropriate I2C Resource Manager
     uint32_t i2c_sda = pinmap_peripheral(sda, PinMap_I2C_SDA);
     uint32_t i2c_scl = pinmap_peripheral(scl, PinMap_I2C_SCL);
-    int ownerID = pinmap_merge(i2c_sda, i2c_scl);
+    uint32_t peripheral = pinmap_merge(i2c_sda, i2c_scl);
+    CORE_UTIL_ASSERT(peripheral != (uint32_t)NC);
+    if (peripheral == (uint32_t)NC) {
```

Comment:
Won't this work differently in release and debug modes because of the `assert` above this line?